File: tests/fixtures/agent_generated/agent_weak_random.ts
// Agent-generated: AI assistant uses Math.random() for security tokens
// Expected: SEC-006 (weak_randomness_ts)

import express from "express";

const app = express();

function generateApiKey(): string {
  // AI generated: Math.random() for API key generation
  const chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
  let result = "";
  for (let i = 0; i < 32; i++) {
    result += chars.charAt(Math.floor(Math.random() * chars.length));
  }
  return result;
}

function generateResetToken(): string {
  // AI generated: Math.random() for password reset token
  return Math.random().toString(36).substring(2, 15) +
    Math.random().toString(36).substring(2, 15);
}

app.post("/api/keys", (req, res) => {
  const key = generateApiKey();
  res.json({ apiKey: key });
});

app.post("/api/reset-password", (req, res) => {
  const token = generateResetToken();
  res.json({ resetToken: token });
});

File: tests/fixtures/agent_generated/agent_weak_random.py
# Agent-generated: AI assistant uses random module for security-sensitive tokens
# Expected: SEC-006 (weak_randomness)

import random
import string
import hashlib
from flask import Flask, jsonify, request

app = Flask(__name__)

def generate_api_key():
    """Generate API key using non-cryptographic random."""
    chars = string.ascii_letters + string.digits
    return ''.join(random.choice(chars) for _ in range(32))

def generate_reset_token():
    """Generate password reset token — weak randomness."""
    return hashlib.md5(str(random.randint(0, 999999)).encode()).hexdigest()

def generate_session_id():
    """Generate session ID with random module."""
    return ''.join(random.choices(string.hexdigits, k=64))

@app.route("/api/keys", methods=["POST"])
def create_api_key():
    key = generate_api_key()
    return jsonify({"api_key": key})

@app.route("/api/password-reset", methods=["POST"])
def password_reset():
    token = generate_reset_token()
    return jsonify({"reset_token": token})
